zingolib/src/wallet/notes/transparent.rs

```diff
@@ -0,0 +1,167 @@
+use std::io::Write;
+
+use byteorder::{ReadBytesExt, WriteBytesExt};
+
+use zcash_primitives::transaction::{components::OutPoint, TxId};
+
+use super::NoteInterface;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct TransparentNote {
+    pub address: String,
+    pub txid: TxId,
+    pub output_index: u64,
+    pub script: Vec<u8>,
+    pub value: u64,
+
+    spent: Option<(TxId, u32)>, // If this utxo was confirmed spent Todo: potential data incoherence with unconfirmed_spent
+
+    // If this utxo was spent in a send, but has not yet been confirmed.
+    // Contains the txid and height at which the Tx was broadcast
+    pub unconfirmed_spent: Option<(TxId, u32)>,
+}
+
+impl NoteInterface for TransparentNote {
+    fn spent(&self) -> &Option<(TxId, u32)> {
+        &self.spent
+    }
+    fn spent_mut(&mut self) -> &mut Option<(TxId, u32)> {
+        &mut self.spent
+    }
+    fn pending_spent(&self) -> &Option<(TxId, u32)> {
+        &self.unconfirmed_spent
+    }
+    fn pending_spent_mut(&mut self) -> &mut Option<(TxId, u32)> {
+        &mut self.unconfirmed_spent
+    }
+}
+
+impl TransparentNote {
+    pub fn new(
```

Comment:
can we change this to `from_parts` to stay in line with sapling/orchard and also the LRZ format?